presto-main/src/main/java/io/prestosql/testing/DateTimeTestingUtils.java

```diff
@@ -63,7 +64,7 @@ public static SqlTimestampWithTimeZone sqlTimestampWithTimeZoneOf(int precision,
         ZonedDateTime base = ZonedDateTime.of(year, month, day, hour, minute, second, 0, timeZoneKey.getZoneId());
 
         long epochMillis = base.toEpochSecond() * MILLISECONDS_PER_SECOND + nanoOfSecond / NANOSECONDS_PER_MILLISECOND;
-        int picosOfMilli = (int) scaleNanosToPicos(nanoOfSecond);
+        int picosOfMilli = (int) (scaleNanosToPicos(nanoOfSecond) % PICOSECONDS_PER_MILLISECOND);
```

Comment:
I dont understand the % here. If the value can `>= PICOSECONDS_PER_MILLISECOND`,  it seems we're discarding some information without compensating it (with eg `epochMillis++` or whatever is appropriate)

Reply:
The new code is actually correct. It works because of the line above. `epochMillis` is computed by taking into account `nanoOfSecond / NANOSECONDS_PER_MILLISECOND`. The remainder is stored in `picosOfMilli`.

It would actually be clearer to do it the other way around. I'll update it.

```java
long epochMillis = base.toEpochSecond() * MILLISECONDS_PER_SECOND + nanoOfSecond / NANOSECONDS_PER_MILLISECOND;
int picosOfMilli = (nanoOfSecond % NANOSECONDS_PER_MILLISECOND) * PICOSECONDS_PER_NANOSECOND;
```

Reply:
yeah, that's so clear that even i can understand. thanks